wai-middleware-auth.cabal

```diff
@@ -0,0 +1,88 @@
+name:                wai-middleware-auth
+version:             0.1.0.0
+synopsis:            Authentication middleware that secures WAI application
+description:         See README
+license:             MIT
+license-file:        LICENSE
+author:              Alexey Kuleshevich
+maintainer:          alexey@fpcomplete.com
+category:            Web
+build-type:          Simple
+extra-doc-files:     README.md CHANGELOG.md
+cabal-version:       >=1.10
+
+library
+  exposed-modules:     Network.Wai.Middleware.Auth
+                       Network.Wai.Middleware.Auth.OAuth2
+                       Network.Wai.Middleware.Auth.OAuth2.Github
+                       Network.Wai.Middleware.Auth.OAuth2.Google
+                       Network.Wai.Middleware.Auth.Provider
+                       Network.Wai.Middleware.Redirect2tls
+                       Network.Wai.Auth.Executable
+  other-modules:       Paths_wai_middleware_auth
+                       Network.Wai.Auth.AppRoot
+                       Network.Wai.Auth.Config
+                       Network.Wai.Auth.ClientSession
+                       Network.Wai.Auth.Tools
+  build-depends:       aeson >= 1.0.2.1
+                     , base              >= 4.7 && < 5
+                     , base64-bytestring >= 1.0
+                     , binary            >= 0.7
+                     , blaze-builder     >= 0.3
+                     , blaze-html
+                     , bytestring        >= 0.10
+                     , case-insensitive  >= 1.2
+                     , cereal
+                     , clientsession     >= 0.9
+                     , containers        >= 0.5
+                     , cookie            >= 0.4
+                     , hoauth2
+                     , http-client       >= 0.4
+                     , http-client-tls   >= 0.2
+                     , http-conduit >= 2.2.3
+                     , http-reverse-proxy
+                     , http-types        >= 0.8
+                     , resourcet         >= 1.1
+                     , safe-exceptions
+                     , shakespeare
+                     , text              >= 1.1
+                     , time              >= 1.4
+                     , unix-compat       >= 0.4
+                     , unordered-containers
+                     , vault
+                     , wai               >= 3.0 && < 4
+                     , wai-app-static
+                     , wai-extra
+                     , yaml
+  hs-source-dirs:      src
+  default-language:    Haskell2010
+  ghc-options:        -Wall
+
+executable wai-auth
+  default-language:    Haskell2010
+  hs-source-dirs:      app
+  main-is:             Main.hs
+  build-depends:       base
```

Comment:
Are all of these deps needed? I thought there were very few imports in the executable itself.